new-frontiers-14/frontier-station-14
# Add FuelVend vendor.

## About the PR



Adds the FuelVend vendor, which sells ship fuel and welding fuel.  The machine's supply is infinite, but the stock is expensive.  It is currently left unmapped.

Adds support for infinite-stock vending machines.

## Why / Balance


Players getting stuck on Frontier Outpost without a source of fuel.

Note: fuel prices still need work.  FuelVend should be a fixed markup (~30% for sake of argument) on crates, and crates should be appropriately priced to how much energy (N.B. _not_ power) they can generate.

## How to test


1. Build and run local server.
2. Spawn in FuelVend vendor.
3. Note the inventory prices.
4. Check that the deny state exists.
5. Note the pretty little rocket animation when you purchase fuel.
6. Note the snarky comments as you leave.
7. Try to open the panel, smash up the machine.
8. Try to unwrench the machine, it shouldn't work.
9. Warp to the Cargo Depot.
10. Spawn a stack of uranium, sell it, it should sell for $954.
11. Spawn a stack of plasma, sell it, it should sell for $954.
12. Spawn a jug of welding fuel, sell it, it should sell for $15.
13. Spawn a jar of AME fuel, sell it, it should sell for $636.
14. Compare all sale prices with the prices in the vendor, the vendor prices should be much higher.

## Media


![image](https://github.com/new-frontiers-14/frontier-station-14/assets/166147148/84b5db63-cd2e-4ba7-a478-533756eceb0d)


## Breaking changes


**Changelog**



:cl:
- add: A FuelVend machine has been added to Frontier Outpost.
- tweak: Tools and flatpacks now have unlimited inventory.
